### PR TITLE
feat: readd support for the custom Data toolbox category

### PR DESCRIPTION
**Pull request by gonfunko**
This PR reintroduces Scratch's custom Data (variables/list) toolbox category. Future PRs will load this category into the toolbox definition in place of the default Blockly Variables category.

### CODE REVIEW

File: src/data_category.js [hunk 1]
```diff
@@ -147,73 +142,61 @@ Blockly.DataCategory.addChangeVariableBy = function (xmlList, variable) {
...
  * @param {?Blockly.VariableModel} variable Variable to select in the field.
  */
-Blockly.DataCategory.addDataList = function (xmlList, variable) {
+function addDataList(xmlList, variable) {
   // <block id="variableId" type="data_listcontents">
   //    <field name="LIST">variablename</field>
   // </block>
-  Blockly.DataCategory.addBlock(xmlList, variable, "data_listcontents", "LIST");
+  addBlock(xmlList, variable, "data_listcontents", "LIST");
   // In the flyout, this ID must match variable ID for monitor syncing reasons
```
Comment (BeksOmega) [inline src/data_category.js:190]: Do you know what this means? Just curious.

File: src/data_category.js [hunk 2]
```diff
@@ -476,7 +444,7 @@ Blockly.DataCategory.addCreateButton = function (xmlList, workspace, type) {
  * @param {?Array.<string>} opt_secondValue Optional array containing the value
  *     name and shadow type of a second pair of value tags.
  */
-Blockly.DataCategory.addBlock = function (
+function addBlock(
```
Comment (BeksOmega) [inline src/data_category.js:447]: You could probably get rid of this whole thing and convert the toolbox category to use JSON if you want instead. It's fine to mix dynamic categories that use JSON with static categories that use XML.

Up to you whether you think the time investment is worth the code cleanliness

**Review by BeksOmega — APPROVED** [review]

File: src/data_category.js (hunk 1 above)
Comment (gonfunko) [reply src/data_category.js:190]: "monitor syncing" here I assume means displaying the value of the variable in the little monitor chip that appears on the stage if you check the box next to the block, presumably it needs the IDs to match to allow for that.

File: src/data_category.js (hunk 2 above)
Comment (gonfunko) [reply src/data_category.js:447]: That's a good idea, but I think it can wait for a cleanup pass once we're feature complete. Opened #89 to track.

**Pull request merged**